### ISSUE TITLE
Response object for list(arrays) changed between google-protobuf@3.21.2 and 3.21.4

### ISSUE THREAD

**Issue by DaveO-Home** [protocolbuffers/protobuf-javascript#226]
See discussion: https://github.com/quarkusio/quarkus/issues/44844

**Comment by umitkucuk** [comment]
Hello, I’m not sure if this is related to the topic, but I’d like to share the issue we’re experiencing.

In our frontend application, we are using `google-protobuf v3.21.2` and `grpc-web v1.5.0`. Since we are using the google-protobuf package with a caret version (^3.21.2), our application started using v3.21.4 when a new patch was released, and it stopped being able to handle the gRPC connection data. When we examined how the data was coming in, we encountered the following result:

using google-protobuf v3.21.2:

```js
{
    "wrappers_": null,
    "arrayIndexOffset_": -1,
    "array": [
        "PING",
        null,
        "PING",
        "541e0fa7-8e28-4f41-bf4c-64ecd995420e"
    ],
    "pivot_": 1.7976931348623157e+308,
    "convertedPrimitiveFields_": {}
}
```

using google-protobuf v3.21.4:

```js
{
    "f": null,
    "D": -1,
    "u": [
        "PING",
        null,
        "PING",
        "7ac9a70f-7136-49fe-8595-33c7306eaced"
    ],
    "G": 1.7976931348623157e+308,
    "a": {}
}
```

It seems that there was probably a breaking change in the patch version :(

We are currently using v3.21.2, but since the latest bundle size improvement, we want to use the latest version. Should we use `u` key instead of `array`?

**Comment by DaveO-Home** [comment]
@umitkucuk I upgraded to v3.21.4 and changed the code to:
```
       /*
          Relaying on data to be at the 4th position in the response object
        */
        const data = Object.keys(response)[3];
```
And then processed the data object. I would not use `u` as it will likely change in the future.  This was a very bad upgrade from the google maintainers. For some reason they think that `getters` are produced??

**Comment by dibenede** [comment]
We produce getters as documented here: https://github.com/protocolbuffers/protobuf-javascript/blob/main/docs/index.md#singular-message-fields

You should not rely on the internal structure of our messages.

**Comment by DaveO-Home** [comment]
@dibenede - `import "google/protobuf/any.proto";` seems to be a new feature? At least the versioning should be at least v3.22.0. As @umitkucuk mentioned above  he's using `^3.21.2`. That way we can expect a breaking change.

**Comment by dibenede** [comment]
@umitkucuk's post refers to the internal structure of our messages. You cannot rely on the naming of our internal fields -- even if there is no minification.

We would expect to bump a major/minor version for changes to [API](https://github.com/protocolbuffers/protobuf-javascript/blob/main/docs/index.md#singular-message-fields), not these internal aspects.

Any protos are a "well known" type (I didn't see where that's coming into play here) with their own API nuances. They're not a new feature for us.  Is the confusion that we only recently documented their usage?

**Comment by DaveO-Home** [comment]
I totally agree that the internal structure should not be used. I was not aware of `Any` and will try it out. However, I can't see the benefit of the compression since to data is completely internal to the client.

**Comment by DaveO-Home** [comment]
@umitkucuk FYI, by adding the `binary` keyword to your protoc config you will get the `getters`.  e.g.
```
--js_out=import_style=commonjs,binary:.......
```
```
const list = response.getYourDefinedList();
```
